subsystems/TypeHandlerLibrary/src/main/java/org/terasology/persistence/typeHandling/coreTypes/GenericMapTypeHandler.java

```diff
@@ -61,26 +61,45 @@ private PersistedData serializeEntry(Map.Entry<K, V> entry, PersistedDataSeriali
 
     @Override
     public Optional<Map<K, V>> deserialize(PersistedData data) {
-        if (!data.isArray()) {
+        if (!data.isArray() || data.isValueMap()) {
+            logger.warn("Incorrect map format detected: object instead of array.\n" +
+                    "Expected\n" +
+                    "  \"mapName\": [\n" +
+                    "    { \"key\": \"...\", \"value\": \"...\" }\n" +
+                    "  ]\n" +
+                    "but found \n'{}'", data);
```

Comment:
Used in multiple places, maybe create a helper method to be used here and below.

```suggestion
                    getUsageInfo(data));
```

```java
private String getUsageInfo(PersistedData data) {
	return "Expected\n" +
                    "  \"mapName\": [\n" +
                    "    { \"key\": \"...\", \"value\": \"...\" }\n" +
                    "  ]\n" +
                    "but found \n'" + data + "'";
}
```

Reply:
done :heavy_check_mark: